### ISSUE TITLE
Checkpointing Fails for ER_ACE

### ISSUE THREAD

**Issue by wistuba** [ContinualAI/avalanche#1268]
🐛 **Describe the bug**
When combining ER_ACE with checkpointing, I get `TypeError: cannot pickle 'generator' object` when writing the checkpoint (avalanche/training/plugins/checkpoint.py:248). This does not happen with Naive, I haven't tried other strategies.

🐜 **To Reproduce**
Based on the example in the README:
```
import torch
from torch.nn import CrossEntropyLoss
from torch.optim import SGD

from avalanche.benchmarks.classic import PermutedMNIST
from avalanche.models import SimpleMLP
from avalanche.training import ER_ACE
from avalanche.training.plugins.checkpoint import CheckpointPlugin, FileSystemCheckpointStorage


# Config
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
# model
model = SimpleMLP(num_classes=10)

# CL Benchmark Creation
perm_mnist = PermutedMNIST(n_experiences=3)
train_stream = perm_mnist.train_stream
test_stream = perm_mnist.test_stream

# Prepare for training & testing
optimizer = SGD(model.parameters(), lr=0.001, momentum=0.9)
criterion = CrossEntropyLoss()

checkpoint_plugin = CheckpointPlugin(
    FileSystemCheckpointStorage(
        directory="checkpoints",
    ),
)

# Continual learning strategy
cl_strategy = ER_ACE(
    model, optimizer, criterion, train_mb_size=32, train_epochs=2, mem_size=200, batch_size_mem=32,
    eval_mb_size=32, device=device, plugins=[checkpoint_plugin])

# train and test loop over the stream of experiences
results = []
for train_exp in train_stream:
    cl_strategy.train(train_exp)
    results.append(cl_strategy.eval(test_stream))
```

**Comment by AlbinSou** [comment]
Hi @wistuba, thanks for reporting the issue

This must be due to the fact that the ER_ACE plugin, unlike the normal replay plugin, keeps a separate dataloader for the replay data. But this should not be needed in the checkpoint since just the storage method is needed to create this dataloader. I guess it just needs one method to tell the checkpointing not to pickle the dataloader. I don't know yet how checkpointing is done but if I have time I will fix this. Otherwise wait for the answer from the authors of the checkpointing plugins.

**Comment by AntonioCarta** [comment]
> I guess it just needs one method to tell the checkpointing not to pickle the dataloader. 

I already added this is my private branch. I will push it as soon as it's ready. As you say, we need to remove problematic attributes before serialization.

**Comment by AntonioCarta** [comment]
Fixed by #1338
